director/director_api.go

```diff
@@ -140,7 +141,7 @@ func VerifyDirectorSDToken(strToken string) (bool, error) {
 	scopes := strings.Split(scope, " ")
 
 	for _, scope := range scopes {
-		if scope == "pelican.directorSD" {
```

Comment:
I think there's a common pattern of splitting the scope string and then seeing the scope is included.  How about writing a helper function in the `TokenScope` class for that?  Something like (untested code):

```
func (target *TokenScope) HasScope(scopeClaim string) bool {
  for _, scope := range strings.Split(scopeString, " ") {
    if target.String() == scope {
      return true
    }
  }
  return false
}
```

Reply:
We already have this function in our codebase and it's even more versatile: https://github.com/PelicanPlatform/pelican/blob/1a47c0f35b69f75950954b81977d2bda39b2c800/utils/server_auth.go#L89

However, I think it's time to migrate this function to `token_scopes` package and export it there, so that things are more organized. Also, since `CheckAnyAuth` is already competent in checking scopes, once we finish #559, we should be good to go.

Reply:
Ok -- we'll do this change in the follow-up (tweaked the title of #559 to reflect this would also handle the token verification case pointed out here).